### ISSUE TITLE
Make Faraday request to Symphony CGI be Asynchronous

### ISSUE THREAD

**Issue by jgreben** [sul-dlss/libsys-webforms#233]
It takes a long time for some Endow funds and Shelf Sel reports to complete. Consider making the request asynchronous.

**Comment by jgreben** [comment]
I think this should be done sooner than later. The latency can be quite long.

**Comment by tallenaz** [comment]
Probably relevant: http://edgeguides.rubyonrails.org/active_job_basics.html

**Comment by tallenaz** [comment]
Also: https://github.com/sul-dlss/DeveloperPlaybook/blob/master/best-practices/background_processing.md

**Comment by tallenaz** [comment]
Installed redis: https://github.com/sul-dlss/puppet/commit/a3af0ccf2bc06c778fdcfb518f9e31bb6a4289b7

**Comment by tallenaz** [comment]
I have a `sidekiq` branch with an issue where Sidekiq can't find the method I want it to kick off.

I've created a worker for the endowed funds report:

```
class EndowedFundsWorker
  include Sidekiq::Worker
  def perform(params)
    submit_endow_funds(params)
  end
end
```

This worker gets called from `EndowedFundsReportsController`:

```
after_action only: :create do
    EndowedFundsWorker.perform_async(batch_params)
  end
```

But Sidekiq is saying: 

```
NoMethodError: undefined method `submit_endow_funds' for #<EndowedFundsWorker:0x0000000516ec98>
```

**Comment by cbeer** [comment]
Two questions:

Where is `submit_endow_funds` defined?

Are there features you need from sidekiq that you can't get from activejob? Using the ActiveJob abstraction will make it easier to switch backends for different environments, making developing with it so much easier.

**Comment by jgreben** [comment]
@tallenaz maybe `include SymphonyCgi` or `include EndowFundsParams` at the start of the EndowedFundsWorker class?

**Comment by tallenaz** [comment]
`submit_endow_funds` is defined in the `EndowFundsParams` module at `app/controllers/concerns/endow_funds_params.rb`. I tried including both `SymphonyCgi` and `EndowFundsParams` in the worker, and am getting the same `NoMethodError`.

I'll look into `activejob`, I don't think there's anything extra we need from sidekiq.

**Comment by jgreben** [comment]
Closed by #261
